### PR TITLE
Timespan collection pages

### PR DESCRIPTION
**Pull request by alansutherland**

### CODE REVIEW

File: middleware/legacy/rules/explore.js [hunk 1]
```diff
@@ -1,7 +1,7 @@
 // Redirect legacy gallery page URLs
 
 export default (route) => {
-  const pattern = /^\/portal(\/[a-z]{2})?\/explore\/(newcontent|colours|sources|topics|people|periods)(\.html)?$/;
+  const pattern = /^\/portal(\/[a-z]{2})?\/explore\/(newcontent|colours|sources|topics|times|people|periods)(\.html)?$/;
```
Comment (lbiedinger) [inline middleware/legacy/rules/explore.js:4]: this won't be needed as there were never 'times' pages on the classic portal

File: middleware/legacy/rules/entity.js [hunk 2]
```diff
@@ -1,13 +1,13 @@
 // Redirect legacy entity page URLs
 
 export default (route) => {
-  const pattern = /^\/portal(\/[a-z]{2})?(\/explore\/(people|topics))(\/[0-9]+)/;
+  const pattern = /^\/portal(\/[a-z]{2})?(\/explore\/(people|topics|times))(\/[0-9]+)/;
```
Comment (rwd) [inline middleware/legacy/rules/entity.js:4]: This should not add "times" but "periods" which is what the classic portal used.

File: middleware/legacy/rules/entity.js [hunk 3]
```diff
@@ -1,13 +1,13 @@
...
   const match = route.path.match(pattern);
 
   return match ? {
     path: [
       match[1],
-      match[2].replace('/explore', '/collections').replace('/people', '/person').replace('/topics', '/topic'),
+      match[2].replace('/explore', '/collections').replace('/people', '/person').replace('/topics', '/topic').replace('/times', '/timespan'),
```
Comment (rwd) [inline middleware/legacy/rules/entity.js:10]: This should replace "periods" with "time" (not "times" with "timespan")

File: plugins/europeana/record.js [hunk 4]
```diff
@@ -202,9 +202,9 @@ export default (axiosOverrides) => {
       const concepts = (edm.concepts || []).map(reduceEntity).map(Object.freeze);
       const places = (edm.places || []).map(reduceEntity).map(Object.freeze);
       const agents = (edm.agents || []).map(reduceEntity).map(Object.freeze);
-      const timespans = (edm.timespans || []).map(reduceEntity).map(Object.freeze);
+      const times = (edm.timespans || []).map(reduceEntity).map(Object.freeze);
```
Comment (rwd) [inline plugins/europeana/record.js:205]: Revert this change, as here we use agents and concepts (as opposed to people and topics) so should also use timespans.

File: plugins/europeana/record.js [hunk 5]
```diff
@@ -202,9 +202,9 @@ export default (axiosOverrides) => {
...
 
-      const entities = [].concat(concepts, places, agents, timespans)
+      const entities = [].concat(concepts, places, agents, times)
```
Comment (rwd) [inline plugins/europeana/record.js:207]: Revert this change, as here we use agents and concepts (as opposed to people and topics) so should also use timespans.

File: plugins/europeana/record.js [hunk 6]
```diff
@@ -223,6 +223,7 @@ export default (axiosOverrides) => {
         media: this.aggregationMedia(providerAggregation, edm.type, edm.services),
         agents,
         concepts,
+        times,
```
Comment (rwd) [inline plugins/europeana/record.js:226]: Here we return agents and concepts (as opposed to people and topics) so should also return timespans (as opposed to times).

**Review by rwd — CHANGES_REQUESTED** [review]

**Review by rwd — APPROVED** [review]

**Pull request merged**